### PR TITLE
Upgrade php & phpunit

### PR DESCRIPTION
**Pull request by funivan**
#### What?
Use latest phpunit 10.  phpunit 9 is End of bugfix soon
image

Use php 8.1 as minimum required php version. php 8.0 is EOL
image

Add rector tool as dev dependency to upgrade test classes

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -4,11 +4,9 @@ composer.lock
...
 # Testing
-/phpunit.xml
 /build
 /test/reports
 .env
 .idea
 /.php_cs
-/.php_cs.cache
-/.phpunit.result.cache
+*.cache
```
Comment (filakhtov) [inline .gitignore:12]: should we use `/*.cache` just to be safe?

File: rector.php [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+use Rector\Config\RectorConfig;
+use Rector\PHPUnit\Set\PHPUnitSetList;
+use Rector\Set\ValueObject\LevelSetList;
+
+return static function (RectorConfig $rectorConfig): void {
+    $rectorConfig->paths([
+        __DIR__ . '/test',
+    ]);
+
+    $rectorConfig->sets([PHPUnitSetList::PHPUNIT_100,]);
```
Comment (filakhtov) [inline rector.php:15]: 🍹 
```suggestion
    $rectorConfig->sets([PHPUnitSetList::PHPUNIT_100]);
```
or
```suggestion
    $rectorConfig->sets([
        PHPUnitSetList::PHPUNIT_100,
    ]);
```

File: .gitignore (hunk 1 above)
Comment (funivan) [reply .gitignore:12]: Yep, let me change it.

**Review by filakhtov — APPROVED** [review]

File: rector.php [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+
+declare(strict_types=1);
+
+use Rector\CodeQuality\Rector\Class_\InlineConstructorDefaultToPropertyRector;
+use Rector\Config\RectorConfig;
+use Rector\PHPUnit\Set\PHPUnitSetList;
+use Rector\Set\ValueObject\LevelSetList;
+
+return static function (RectorConfig $rectorConfig): void {
+    $rectorConfig->paths([__DIR__ . '/test',]);
```
Comment (filakhtov) [inline rector.php:11]: Now this is mis-formatted 😅

**Review by filakhtov — APPROVED** [review]

File: rector.php (hunk 3 above)
Comment (funivan) [reply rector.php]: I have removed trailing comma =)

**Review by repzy — APPROVED** [review]

**Pull request merged**